index.js

```diff
@@ -26,5 +60,130 @@ module.exports = {
         usePrettierrc: false,
       },
     ],
+
+    // We use TypeScript so don't usually need PropTypes.
+    'react/prop-types': 'off',
+  },
+
+  // TypeScript-specific rules. Somewhat inspired by tslint:recommended.
+  overrides: {
+    files: ['**/*.ts', '**/*.tsx'],
+    parser: '@typescript-eslint/parser',
+    parserOptions: {
+      ecmaVersion: 2018,
+      sourceType: 'module',
+      ecmaFeatures: {
+        jsx: true,
+      },
+    },
+
+    rules: {
+      // Require overloads to be grouped together
+      '@typescript-eslint/adjacent-overload-signatures': 'error',
+
+      // Use T[] over Array<T> for simple types
+      '@typescript-eslint/array-type': ['error', 'array-simple'],
+
+      // Prevent use of certain types that are almost always mistakes.
+      '@typescript-eslint/ban-types': [
+        'error',
+        {
+          types: {
+            Boolean: {
+              fixWith: 'boolean',
+              message:
+                'Avoid using the `Boolean` type. Did you mean `boolean`?',
+            },
+            Function: {
+              message:
+                'Avoid using the `Function` type. Prefer a specific function type like `() => void`.',
+            },
+            Number: {
+              fixWith: 'number',
+              message: 'Avoid using the `Number` type. Did you mean `number`?',
+            },
+            Object: {
+              message: 'Avoid using the `Object` type. Did you mean `object`?',
+            },
+            String: {
+              fixWith: 'string',
+              message: 'Avoid using the `String` type. Did you mean `string`?',
+            },
+            Symbol: {
+              fixWith: 'symbol',
+              message: 'Avoid using the `Symbol` type. Did you mean `symbol`?',
+            },
+          },
+        },
+      ],
+
+      // Require camelCase names, except for property names.
+      camelcase: 'off',
+      '@typescript-eslint/camelcase': ['error', { properties: 'never' }],
```

Comment:
Is the exception for property names to allow for constructing objects/types that map to our APIs which use snake case?

Reply:
Yep, exactly